pdflayers/utils.py

```diff
@@ -7,26 +7,34 @@
 logger = logging.getLogger(__name__)
 
 
-def set_layer_visibility(pdf, layers_to_show):
+def set_layers(pdf, layers_to_show, layers_to_remove):
     """Set visibility of layers."""
     try:
         ocgs = pdf.root.OCProperties.OCGs
+
     except (AttributeError, KeyError):
         logger.error("Unable to locate layers in PDF.")
         sys.exit(1)
 
     ocgs_on = []
+    ocgs_keep = []
     for ocg in ocgs:
+        """Set visibility/active_state of layers."""
         if ocg.Name in layers_to_show:
             logger.info("Layer %s will be visible.", ocg.Name)
             ocgs_on.append(ocg)
+            ocgs_keep.append(ocg)
         else:
-            logger.info("Layer %s will be hidden.", ocg.Name)
+            if ocg.Name not in layers_to_remove:
+                logger.info("Layer %s will be hidden.", ocg.Name)
+                ocgs_keep.append(ocg)
+            else:
+                logger.info("Layer %s will be removed.", ocg.Name)
 
     ocgs_config = pikepdf.Dictionary(
         BaseState=pikepdf.Name('/OFF'),
         ON=ocgs_on,
-        Order=ocgs,
+        Order=ocgs_keep,
```

Comment:
I need to to check the specification, but since the hidden layers are still there I would expect that you should still list all the layers here

Reply:
I am not sure... since as you may realize. I thought we are removing them... now they are just hidden even from the side bar. Would have been fantastic if we could really remove them.

I started a conversation with qpdf, and there seems to be more to this... I am looking at some of the suggestions that came up in the conversation. But have not had the time to really dig in. I will leave this part as I have it... Don't think it will break anything. But I will update the printed text to say "hidden".